storage/range_command.go

```diff
@@ -344,11 +344,20 @@ func (r *Range) InternalRangeLookup(batch engine.Engine, args *proto.InternalRan
 	// We want to search for the metadata key just greater than args.Key. Scan
 	// for both the requested key and the keys immediately afterwards, up to
 	// MaxRanges.
-	metaPrefix := proto.Key(args.Key[:len(engine.KeyMeta1Prefix)])
-	nextKey := proto.Key(args.Key).Next()
+	var metaPrefix proto.Key
+	var startKey proto.Key
+	if len(args.Key) > 0 {
```

Comment:
Did changing engine.RangeMetaKey to return engine.KeyMeta1Prefix+KeyMin if supplied KeyMin not work?

Also, let's move this code into engine.

    func DecodeRangeMetaKey(key proto.Key) (proto.Key, proto.Key)